multiqc/modules/fastqc/fastqc.py

```diff
@@ -93,41 +101,62 @@ def __init__(self):
         # Add to the general statistics table
         self.fastqc_general_stats()
 
-        # Add the statuses to the intro for multiqc_fastqc.js JavaScript to pick up
+        # Add the statuses to the intro for multiqc_fastqc.js JavaScript to pick up.
+        # FIXME - if the graphs want to show pass/fail bar for just read1 or read2 this
+        # is going to take some JS hacking, but just fudge it for now - the call
+        # will be based on whichever read is seen last!
         statuses = dict()
-        for s_name in self.fastqc_data:
-            for section, status in self.fastqc_data[s_name]['statuses'].items():
+        for (s_name, read_num), s_fastqc_data in self.fastqc_data.items():
+            for section, status in s_fastqc_data['statuses'].items():
                 try:
                     statuses[section][s_name] = status
                 except KeyError:
                     statuses[section] = {s_name: status}
         self.intro += '<script type="text/javascript">fastqc_passfails = {};</script>'.format(json.dumps(statuses))
 
         # Now add each section in order
-        self.sequence_quality_plot()
-        self.per_seq_quality_plot()
-        self.sequence_content_plot()
-        self.gc_content_plot()
-        self.n_content_plot()
-        self.seq_length_dist_plot()
-        self.seq_dup_levels_plot()
-        self.overrepresented_sequences()
-        self.adapter_content_plot()
-
-    def parse_fastqc_report(self, file_contents, s_name=None, f=None):
+        # Allow plot selection in config
+        try:
+            plots_enabled = config.fastqc_plots_enabled
```

Comment:
```python
plots_enabled = getattr(config, 'fastqc_plots_enabled', [])
```

Reply:
This is something I added just because I needed it myself (sorry there is a lot of conflated stuff in this PR!). Do you like the idea for inclusion in mainline? In general, should modules be allowed to have their own configuration and if so should it look like this?

Reply:
Yup, definitely. Quite a few modules already have their own config - eg. [preseq](https://github.com/ewels/MultiQC/blob/master/multiqc/modules/preseq/preseq.py#L78-L80), [qualimap](https://github.com/ewels/MultiQC/blob/master/multiqc/modules/qualimap/QM_BamQC.py#L20-L30) and others.

In fact, FastQC is already doing it [here](https://github.com/ewels/MultiQC/blob/master/multiqc/modules/fastqc/fastqc.py#L447), so maybe nest under the `fastqc_config` key?

Reply:
Ah - how did I miss that? Fixed (and noted that it needs documenting)